src/main/kotlin/org/taktik/icure/be/format/logic/impl/KmehrReportLogicImpl.kt

```diff
@@ -108,12 +111,16 @@ class KmehrReportLogicImpl : GenericResultFormatLogicImpl(), KmehrReportLogic {
 				val protocolId = t.ids.find { it.s == IDKMEHRschemes.ID_KMEHR }?.value
 				val demandTimestamp = demandEpochMillis(t)
 
-				val s = Service().apply {
-					id = uuidGen.newGUID().toString()
-					content.put(language, Content(t.headingsAndItemsAndTexts.filterIsInstance(TextType::class.java).joinToString(separator = "\n") { it.value }))
-					label = "Protocol"
-					demandTimestamp?.let { valueDate = FuzzyValues.getFuzzyDate(LocalDateTime.ofInstant(Instant.ofEpochMilli(it), ZoneId.systemDefault()), ChronoUnit.SECONDS) }
-				}
+                var s: Service? = null;
+                val textItems = t.headingsAndItemsAndTexts.filterIsInstance(TextType::class.java);
+                if (textItems.isNotEmpty()) {
```

Comment:
use if expression with `else null` instead of using `var s: Service? = null`: 
`val s: Service? = if(...) { ... } else null`, also here type definition can be omitted 
also, semicolon can be omitted :)